test/cpp/jit/test_gpu.cpp

```diff
@@ -4605,6 +4605,90 @@ void testGPU_FusionReductionScheduler() {
       aten_output.sub(cg_output).abs().max());
 }
 
+void testGPU_FusionReductionSchedulerMultiDimNonFastest() {
+  const std::vector<int> red_dims = {0, 2};
+  const std::vector<int64_t> red_dims64 = {0, 2};
+  const std::vector<int64_t> tensor_dims_in({5, 10, 15, 20});
+  const std::vector<int64_t> tensor_dims_out = {10, 20};
+
+  torch::jit::fuser::cuda::CudaKernel prog;
+  prog.setFusionPtr(std::make_unique<Fusion>());
```

Comment:
nit: why not `Fusion fusion` or `auto fusion = std::make_unique<Fusion>()`, then move CudaKernel later on, where it's actually used?